android-sdk/src/main/java/com/mobilecoin/lib/MobileCoinClient.java

```diff
@@ -568,7 +568,8 @@ public long submitTransaction(@NonNull Transaction transaction)
                 "transaction:", transaction);
         ConsensusCommon.ProposeTxResponse txResponse =
                 consensusClient.proposeTx(transaction.toProtoBufObject());
-        this.txOutStore.setConsensusBlockIndex(UnsignedLong.fromLongBits(txResponse.getBlockCount() - 1L));
+        final long blockCount = txResponse.getBlockCount() > 0 ? txResponse.getBlockCount() - 1L : 0;
+        this.txOutStore.setConsensusBlockIndex(UnsignedLong.fromLongBits(blockCount));
```

Comment:
```suggestion
        final long blockIndex = txResponse.getBlockCount() > 0 ? txResponse.getBlockCount() - 1L : 0;
        this.txOutStore.setConsensusBlockIndex(UnsignedLong.fromLongBits(blockIndex));
```
This is good, but let's rename the variable here to blockIndex because it represents not the count, but the index

Reply:
Ah, that's good to know.